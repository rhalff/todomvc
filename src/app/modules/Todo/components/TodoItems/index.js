// @flow
import React, { PureComponent } from 'react'
import styles from './scss/TodoItems.scss'
import TodoItem from '../TodoItem'
import type { Todos } from '../../types/todos'

type Props = {
  items: Todos,
  onRemoveItem: Function,
  onSaveItem: Function,
  onToggleItemComplete: Function
}

class TodoItems extends PureComponent<Props> {
  renderItems = () => {
    const { items, onRemoveItem, onSaveItem, onToggleItemComplete } = this.props

    return items.map(item => (
      <li key={item.id}>
        <TodoItem
          item={item}
          onRemove={onRemoveItem}
          onSave={onSaveItem}
          onToggleComplete={onToggleItemComplete}
        />
      </li>
    ))
  }

  render() {
    return (
      <section className={styles.main}>
        <ul className={styles.todoItems}>{this.renderItems()}</ul>
      </section>
    )
  }
}

export default TodoItems
