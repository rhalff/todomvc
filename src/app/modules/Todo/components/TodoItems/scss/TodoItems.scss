@import 'colors';

.todoItems {
  margin: 0 0 30px 25px;
  padding: 0;
  margin: 0;
  list-style: none;

  li {
    position: relative;
    font-size: 24px;
    border-bottom: 1px solid $todoItemBorderColor;

    &.editing {
      border-bottom: none;
      padding: 0;
    }

    &:last-child {
      border-bottom: none;
    }
  }
}

.main {
  position: relative;
  z-index: 2;
  border-top: 1px solid $todoItemBorderColor;
}
