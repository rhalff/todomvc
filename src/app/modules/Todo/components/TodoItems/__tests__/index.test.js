// @flow
import React from 'react'
import TodoItems from '../'
import renderer from 'react-test-renderer'

const todos = [
  { id: 1, text: 'Style & compose all components' },
  { id: 2, text: 'App Init', completed: true },
  { id: 3, text: 'Copy Styles one on one' }
]

test('Can show Todo Items', () => {
  const component = renderer.create(
    <TodoItems items={todos} onSaveItem={() => {}} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
