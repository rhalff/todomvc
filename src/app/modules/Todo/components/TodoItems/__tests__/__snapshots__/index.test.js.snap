// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Todo Items 1`] = `
<section
  className="main"
>
  <ul
    className="todoItems"
  >
    <li>
      <div
        className="active"
      >
        <input
          checked={undefined}
          className="unchecked"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="labelEditor"
        >
          <div
            contentEditable={false}
            onDoubleClick={[Function]}
            onKeyDown={[Function]}
          >
            Style & compose all components
          </div>
        </div>
        <button
          className="removeButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
    </li>
    <li>
      <div
        className="completed"
      >
        <input
          checked={true}
          className="checked"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="labelEditor"
        >
          <div
            contentEditable={false}
            onDoubleClick={[Function]}
            onKeyDown={[Function]}
          >
            App Init
          </div>
        </div>
        <button
          className="removeButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
    </li>
    <li>
      <div
        className="active"
      >
        <input
          checked={undefined}
          className="unchecked"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          className="labelEditor"
        >
          <div
            contentEditable={false}
            onDoubleClick={[Function]}
            onKeyDown={[Function]}
          >
            Copy Styles one on one
          </div>
        </div>
        <button
          className="removeButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
    </li>
  </ul>
</section>
`;
